### PR TITLE
Deterministic retention query behavior

### PR DESCRIPTION
**Pull request by gkorland**
Always filter query result by max retention, this can be done simple by
using max(start_time, seriesMaxTimestamp - retention).

### CODE REVIEW

**Comment by gkorland** [comment]
Fix RED-28975

File: src/module.c [hunk 1]
```diff
@@ -386,6 +386,11 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
         AggregationClass *aggObject, int64_t time_delta) {
     RedisModule_ReplyWithArray(ctx, REDISMODULE_POSTPONED_ARRAY_LEN);
     long long arraylen = 0;
+
+	// In case a retention is set shouldn't return chunks older than the retention
```
Comment (danni-m) [inline src/module.c]: @gkorland tab vs spaces

Comment (gkorland) [reply src/module.c]: fixed

**Comment by gkorland** [comment]
@danni-m any other comments?

**Pull request merged**